h5grove/encoders.py

```diff
@@ -77,12 +78,25 @@ def tiff_encode(data: np.ndarray) -> bytes:
         return buffer.getvalue()
 
 
-class Response(NamedTuple):
+class Response:
     content: Union[Generator[bytes, None, None], bytes]
     """ Encoded `content` as a generator of bytes """
     headers: Dict[str, str]
     """ Associated headers """
 
+    def __init__(
+        self,
+        content: Union[Generator[bytes, None, None], bytes],
+        headers: Dict[str, str],
+    ):
+        self.content = content
+        size = (
+            reduce(lambda acc, c: acc + len(c), content, 0)
```

Comment:
This will consume `content` and the generator will be empty:
```python
In [9]: def gen(): 
   ...:     yield 1 
   ...:     yield 2                                                                                                
In [10]: g = gen()                                                                                     
In [11]: sum(g)                                                                                        
Out[11]: 3
In [12]: sum(g)                                                                                        
Out[12]: 0
```
The only option for `Generator` seems to be providing the `size` to `__init__`.
Not sure it is possible for all formats, but for the previous `npy_stream` encoder this was possible (and I would expect so for any uncompressed format).
We can get rid of the `Generator` for now since I think it is not currently used.